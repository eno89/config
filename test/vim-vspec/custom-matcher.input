call vspec#hint({'scope': 'vspec#_scope()', 'sid': 'vspec#_sid()'})
function! s:describe__s_valid_matcher_custom_p()  "{{{1
  It should return true for "be"

  Should Call('s:valid_matcher_custom_p', 'be') == !0

  It should return false for other value

  Should Call('s:valid_matcher_custom_p', 'foo') == !!0
  Should Call('s:valid_matcher_custom_p', 'being') == !!0
  Should Call('s:valid_matcher_custom_p', 'describe') == !!0
endfunction




function! s:describe__vspec_customize_matcher()  "{{{1
  function! True(value)
    return !!(a:value)
  endfunction
  function! False(value)
    return !(a:value)
  endfunction

  It should succeed

  call vspec#customize_matcher('foo', function('True'))
  Should Ref('s:custom_matchers')['foo'] ==# function('True')

  call vspec#customize_matcher('foo', function('False'))
  Should Ref('s:custom_matchers')['foo'] ==# function('False')
endfunction




" __END__  "{{{1
" vim: filetype=vim foldmethod=marker
