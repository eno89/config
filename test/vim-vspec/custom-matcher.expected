==== s_valid_matcher_custom_p 
---- It should return true for "be" 
.
---- It should return false for other value 
...

==== vspec_customize_matcher 
---- It should succeed 
..



**** Result ****


6 examples, 0 failures
